app/views/LanguageView/index.js

```diff
@@ -0,0 +1,183 @@
+import RocketChat from '../../lib/rocketchat';
+import KeyboardView from '../../presentation/KeyboardView';
+import sharedStyles from '../Styles';
+import RCTextInput from '../../containers/TextInput';
+import scrollPersistTaps from '../../utils/scrollPersistTaps';
+import I18n from '../../i18n';
+import Button from '../../containers/Button';
+import Loading from '../../containers/Loading';
+import { showErrorAlert, showToast } from '../../utils/info';
+import log from '../../utils/log';
+import { setUser as setUserAction } from '../../actions/login';
+import { DrawerButton } from '../../containers/HeaderButton';
+import StatusBar from '../../containers/StatusBar';
+
+@connect(state => ({
+	userLanguage: state.login.user && state.login.user.language
+}), dispatch => ({
+	setUser: params => dispatch(setUserAction(params))
+}))
+/** @extends React.Component */
+export default class LanguageView extends LoggedView {
+	static navigationOptions = ({ navigation }) => ({
+		headerLeft: <DrawerButton navigation={navigation} />,
+		title: I18n.t('Change Language')
+	})
+
+	static propTypes = {
+		componentId: PropTypes.string,
+		userLanguage: PropTypes.string,
+		setUser: PropTypes.func
+	}
+
+	constructor(props) {
+		super('LanguageView', props);
+		this.state = {
+			placeholder: {},
+			language: props.userLanguage ? props.userLanguage : 'en',
+			languages: [{
+				label: 'English',
+				value: 'en'
+			}, {
+				label: 'Português (BR)',
+				value: 'pt-BR'
+			}, {
+				label: 'Russian',
+				value: 'ru'
+			}, {
+				label: '简体中文',
+				value: 'zh-CN'
+			}, {
+				label: 'Français',
+				value: 'fr'
+			}, {
+				label: 'Deutsch',
+				value: 'de'
+			}, {
+				label: 'Português (PT)',
+				value: 'pt-PT'
+			}],
+			saving: false
+		};
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
+		const { language, saving } = this.state;
+		const { userLanguage } = this.props;
+		if (nextState.language !== language) {
+			return true;
+		}
+		if (nextState.saving !== saving) {
+			return true;
+		}
+		if (nextProps.userLanguage !== userLanguage) {
+			return true;
+		}
+		return false;
+	}
+
+	getLabel = (language) => {
+		const { languages } = this.state;
+		const l = languages.find(i => i.value === language);
+		if (l && l.label) {
+			return l.label;
+		}
+		return null;
+	}
+
+	formIsChanged = () => {
+		const { userLanguage } = this.props;
+		const { language } = this.state;
+		return !(userLanguage === language);
+	}
+
+	submit = async() => {
+		this.setState({ saving: true });
+
+		const { language } = this.state;
+		const { userLanguage, setUser } = this.props;
+
+		if (!this.formIsChanged()) {
+			return;
+		}
+
+		const params = {};
+
+		// language
+		if (userLanguage !== language) {
+			params.language = language;
+		}
+
+		try {
+			await RocketChat.saveUserPreferences(params);
+			setUser({ language: params.language });
+
+			this.setState({ saving: false });
+			setTimeout(() => {
+				showToast(I18n.t('Preferences_saved'));
```

Comment:
add logic to navigate to chat stack or some higher level stack after success so that all views can be re-rendered

Reply:
Done

Reply:
but need to review